### PR TITLE
Add reading_time to articles api responses

### LINKED ISSUE #13436: Missing read time from GET articles endpoint
I am building a multiplatform application for DEV that will run on iOS, iPad OS, and macOS using SwiftUI ([here](https://github.com/hadiidbouk/DEV)).
The problem is that [GET articles endpoint](https://docs.forem.com/api/#operation/getArticles) doesn't contain the "read time" property.

The "3 min read" is added just for testing purposes.

Screen Shot 2021-04-09 at 2 54 03 PM

### PR DESCRIPTION
**Pull request by djuber**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

When we serialize articles for the api, we omit reading time. There are applications that could use this.

## Related Tickets & Documents

Fixes #13436

## QA Instructions, Screenshots, Recordings

Exercise the api - expect existing keys, _and_ reading_time, in the json response for the article.

### UI accessibility concerns?

None - api change only

## Added tests?

- [x] Yes (adapted existing tests about response keys to match changes).
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [x] I've updated the [Developer Docs](https://docs.forem.com) and/or
      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] I've updated the README or added inline documentation
- [x] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post 
- [ ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [ ] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] What gif best describes this PR or how it makes you feel?

tl;dr

### CODE REVIEW

File: app/controllers/api/v0/articles_controller.rb [hunk 1]
```diff
@@ -16,7 +16,7 @@ class ArticlesController < ApiController
         title description main_image published_at crossposted_at social_image
         cached_tag_list slug path canonical_url comments_count
         public_reactions_count created_at edited_at last_comment_at published
-        updated_at video_thumbnail_url
+        updated_at video_thumbnail_url reading_time
```
Comment (djuber) [inline app/controllers/api/v0/articles_controller.rb:19]: it might make me medium happy if this were alphabetized but I probably worry about that too much - suffice it to say I did wonder _where_ to add the new attribute to the list for more than a second.

File: docs/api_v0.yml [hunk 2]
```diff
@@ -254,6 +255,10 @@ components:
           format: date-time
         user:
           $ref: "#/components/schemas/SharedUser"
+        reading_time:
```
Comment (djuber) [inline docs/api_v0.yml:258]: I manually edited these items - if this yaml file is machine generated rather than designed for human modification - I probably _didn't_ make upstream changes to the sources and need help.

**Comment by djuber** [comment]
Draft changelog post here https://forem.dev/djuber/added-reading-time-to-articles-api-2e45-temp-slug-5512191?preview=bccf995f6ac1934e7e35f32aca7a881effe75794a033f1f4b2c3d6f259da4f20fd8246e4d54d712a22ef8c7e11b756685c84b479121b31c3af81bf4e

**Review by rhymes — COMMENTED** [review]
Why not! I don't see a reason why not have this exposed in the API, and I can't remember if there was one not to :D

Maybe double check with @mstruve?

File: docs/api_v0.yml (hunk 2 above)
Comment (rhymes) [reply docs/api_v0.yml:258]: This file is human generated, no upstream changes required.

**Comment by mstruve** [comment]
@djuber since this is a feature drop a note about this PR in the #product channel and make sure they are cool with it and if they are 🚀

**Comment by djuber** [comment]
> @djuber since this is a feature drop a note about this PR in the #product channel and make sure they are cool with it and if they are rocket

stamp-of-approval

**Review by mstruve — APPROVED** [review]

**Review by joshpuetz — APPROVED** [review]

**Comment by jgaskins** [comment]
Do we have any standards around communicating time durations via the API? It looks like when we communicate timestamps they're all suffixed with `_at` and in ISO8601/RFC3339 format, durations as raw numbers might be a bit confusing since AFAIK there's no de facto standard for communicating the duration between two timestamps in JSON.

@djuber I see you updated the docs here to specify that the number is minutes, but I wonder if it'd be useful to call that out in the property name itself: `reading_time_minutes`.

**Comment by djuber** [comment]
> @djuber I see you updated the docs here to specify that the number is minutes, but I wonder if it'd be useful to call that out in the property name itself: `reading_time_minutes`.

I have no objections in either case (the fact that it's `reading_time` internally seems an implementation detail). My lazy half definitely prefers not to make the external representation and the internal name differ, but it shouldn't be difficult to do it.  My correctness seeking half sure likes a good argument about adding units where applicable.

**Review by jgaskins — APPROVED** [review]
Thanks for talking that through. I agree 100% with your points about internal vs external representation, and agree that the names we expose are most critical.
